Add notices

danfinlay opened the pull request:
Squashed & cherry picked these changes that were against `dev`.

2-am-zzz committed:
Add ability to show notices to user & get confirmation.

Implement generation of markdown for notice files.
Create npm command. Enhance notice generation.
Add test files to test multiple notices.
Add basic markdown support to notices.
Interval checks for updates.
Add extensionizer and linker
Add terms and conditions state file
Add link support to disclaimer.
Changelog addition.

2-am-zzz committed:
Add ability to show notices to user & get confirmation.

Implement generation of markdown for notice files.
Create npm command. Enhance notice generation.
Add test files to test multiple notices.
Add basic markdown support to notices.
Interval checks for updates.
Add extensionizer and linker
Add terms and conditions state file
Add link support to disclaimer.
Changelog addition.

kumavis reviewed (commented):
por que tienes los dos?

kumavis commented on app/scripts/metamask-controller.js:
```diff
@@ -17,6 +18,12 @@ module.exports = class MetamaskController {
+    })
+    this.noticeController = new NoticeController({
+      configManager: this.configManager,
+    })
```
double notice controller?

danfinlay committed:
Rebuild development states before ci tests

kumavis commented on development/states/notice.json:
```diff
@@ -0,0 +1,63 @@
+  "metamask": {
+    "isInitialized": true,
+    "isUnlocked": true,
+    "rpcTarget": "https://rawtestrpc.metamask.io/",
```
idk if it matters but we should prolly update this

kumavis reviewed (commented):
can we move everything out of config manager and into notif-manager?

2-am-zzz committed:
The first notice, the fall of morden.

2-am-zzz committed:
Clean notices.

2-am-zzz committed:
remove duplicate controller.

kumavis commented:
@Zanibas can we move everything out of config manager and into notif-manager?

danfinlay committed:
Cleanup

danfinlay committed:
Merge branch 'AddNotices' of github.com:MetaMask/metamask-plugin into AddNotices

2-am-zzz committed:
Deactivate polling for now.

kumavis committed:
clean - code style

kumavis committed:
meta - add npm-debug.log to gitignore

kumavis committed:
move notice code from metamask-controller + config-manager, in to notice-controller

2-am-zzz committed:
Add date to notices display.

danfinlay committed:
Clean up first notice

Remove redundant title from notice component.
Make link color the same orange as the rest of the UI.

kumavis reviewed (approved):
😎
